### ISSUE TITLE
Feature request: OAuthServerException extra parameters

### ISSUE THREAD

**Issue by stratoss** [thephpleague/oauth2-server#717]
@alexbilbie would you accept PR adding `$jsonOptions` and `$extraResponseFields` optional variables to `OAuthServerException::generateHttpResponse()`?

The idea is to make the response more configurable, because for one of my implementations I need to append extra fields to the array and use pretty print by default. At the moment I'm catching the body, parsing the JSON, appending the new fields and printing using the extra options which is kinda hacky.

**Comment by alexbilbie** [comment]
I think it'd be better to have a getter and setter method for the `$payload` with which you can inject additional parameters:

```
$exception->setPayload(
  array_merge(
    $exception->getPayload(),
    ['foo' => 'bar']
  )
);
return $exception->generateHttpResponse($response);
```

**Comment by stratoss** [comment]
I agree, what about optional `$jsonOptions` so we can pass `JSON_UNESCAPED_SLASHES | JSON_PRETTY_PRINT` there for example?

**Comment by Sephster** [comment]
Fixed with merge fea577f
